### DISCUSSION TITLE
Failing to install AB on a M2 Macbook

### DISCUSSION THREAD

**Discussion by MarlePr** [LCA-ActivityBrowser/activity-browser#1266] [General]
@haasad Hi Adrian, I try to install the activity browser on a Macbook with a M2 chip (macOS Sonoma 14.4) but I don't know how to fix the reoccurring errors. This is what I've done so far:

1. Install rosetta
2. CONDA_SUBDIR=osx-64 conda create -n ab python=3.9
conda activate ab
conda env config vars set CONDA_SUBDIR=osx-64
3. conda install --override-channels -c haasad/label/experimental -c conda-forge -c
4. conda deactivate
5. conda activate ab
6. (ab) marlene.@18 ~ % activity-browser

Following error occurred:
Traceback (most recent call last):
File "/opt/miniconda3/envs/ab/bin/activity-browser", line 6, in
from activity_browser import run_activity_browser
File "/opt/miniconda3/envs/ab/lib/python3.9/site-packages/activity_browser/init.py", line 6, in
from PySide2.QtCore import QSysInfo, version as qt_version
ImportError: dlopen(/opt/miniconda3/envs/ab/lib/python3.9/site-packages/PySide2/QtCore.cpython-39-darwin.so, 0x0002): Symbol not found: __ZTVNSt3__13pmr25monotonic_buffer_resourceE
Referenced from: <01167352-4F24-347C-8C0A-A347F52A3E34> /opt/miniconda3/envs/ab/lib/libQt5Core.5.15.8.dylib
Expected in: <4EC4E077-EE60-3D48-A7DE-34C55BF3FAB2> /opt/miniconda3/envs/ab/lib/libc++.1.0.dylib

Asked ChatGPT -> said to force install PySide2 with: '(ab) marlene.@18 ~ % pip install --force-reinstall PySide2'

Following error occurred:
Installing collected packages: shiboken2, PySide2
Attempting uninstall: shiboken2
Found existing installation: shiboken2 5.15.7
ERROR: Cannot uninstall shiboken2 5.15.7, RECORD file not found. You might be able to recover from this via: 'pip install --force-reinstall --no-deps shiboken2==5.15.7'.

Also, I have another working venv in which Brightway2 (version2.4.1) and ecoinvent 3.8 cutoff are already set up (thanks to another issue where you and Romain discussed how to do this). Can I install the activity browser in there or is it possible that I mess up my working venv? If I were able use the same venv, how would I install the AB in there?

**Comment by MarlePr** [comment]
And the following packages are currently present in my 'ab' venv:

![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/165831221/0532380b-9d0d-4780-bc1f-1ade17150afa)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/165831221/5215b80f-0f77-4a00-825f-37f66e334cae)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/165831221/f7e986a6-9918-467b-a1a6-ee22ad2ef9e0)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/165831221/54df5e6a-7d26-4e25-9150-77f6e8f2173c)

**Comment by haasad** [comment]
Just to make sure I understand what's going on, is this correct?:
- You have installed the AB following the [installation instructions](https://github.com/LCA-ActivityBrowser/activity-browser?tab=readme-ov-file#the-quick-way)?
```
conda create -n ab -c conda-forge --solver libmamba activity-browser
conda activate ab
activity-browser
```
- What problem/error do you get there?
- You then found issue #741 and tried the solutions mentioned there, but this also didn't work?

Unless there is some problem with M2 Macs that we're not aware of, the normal install instructions should work. No rosetta or other work-arounds needed.

**Comment by MarlePr** [comment]
I'm so sorry. I don't know what I did wrong all day yesterday but today it somehow worked. Thanks a lot!
